### PR TITLE
refactor(deps): bump clap to v4

### LINKED ISSUE #6416: deps: upgrade clap to 4

### PR DESCRIPTION
**Pull request by TennyZhuang**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Bump clap to v4

Close #6416 

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

### Release note

</details>

### CODE REVIEW

**Comment by TennyZhuang** [comment]
Please review `enable_jaeger_tracing` carefully, I'm not sure why we need `Flag` before.

**Review by yuhao-su — APPROVED** [review]
LGTM

File: src/tests/regress/README.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ RUST_BACKTRACE=1 target/debug/risingwave_regress_test -h 127.0.0.1 \
 * Run tests against PostgreSQL. Make sure PostgreSQL is running.
 
 ```shell
-RUST_BACKTRACE=1 target/debug/risingwave_regress_test -h 127.0.0.1 \
+RUST_BACKTRACE=1 target/debug/risingwave_regress_test --host 127.0.0.1 \
```
Comment (BugenZhao) [inline src/tests/regress/README.md:48]: cc @arkbriar

**Review by BugenZhao — APPROVED** [review]

File: src/common/src/config.rs [hunk 2]
```diff
@@ -53,15 +49,6 @@ where
...
 }
 
-/// Map command line flag to `Flag`. Should only be used in `#[derive(OverrideConfig)]`.
-pub fn true_if_present(b: bool) -> Flag {
-    if b {
-        Some(true)
-    } else {
-        None
-    }
-}
```
Comment (Gun9niR) [inline src/common/src/config.rs:63]: If we use default parsing function for `Option<bool>`, the expected cli would be 

image

whereas we simply want `--enable-jaeger-tracing`.

So I think we should bring `true_if_present` back

Comment (Gun9niR) [reply src/common/src/config.rs:63]: nvm, I will fix it 🤣

File: src/tests/regress/README.md (hunk 1 above)
Comment (arkbriar) [reply src/tests/regress/README.md:48]: Thanks for the kind reminder. It seems it won't affect the operator.

**Pull request merged**